From 84f82c95bcd48042437bfd08c77866414c69476f Mon Sep 17 00:00:00 2001
From: Alan Modra <amodra@gmail.com>
Date: Tue, 9 Nov 2021 09:02:22 +1030
Subject: [PATCH 79/97] PR28543, readelf entered an infinite loop

This little tweak terminates fuzzed binary readelf output a little
quicker.

	PR 28543
	* dwarf.c (read_and_display_attr_value): Consume a byte when
	form is unrecognized.
---
 binutils/dwarf.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/binutils/dwarf.c b/binutils/dwarf.c
index d80235fb206..ce54813a076 100644
--- a/binutils/dwarf.c
+++ b/binutils/dwarf.c
@@ -2743,6 +2743,8 @@ read_and_display_attr_value (unsigned long           attribute,
       
     default:
       warn (_("Unrecognized form: 0x%lx\n"), form);
+      /* What to do?  Consume a byte maybe?  */
+      ++data;
       break;
     }
 
-- 
2.34.0

